policies-cn-regions/instance-logging-policy.json

```diff
@@ -0,0 +1,18 @@
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "allowLoggingToCloudWatch",
+      "Effect": "Allow",
+      "Action": [
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents",
+        "logs:DescribeLogStreams"
+      ],
+      "Resource": [
+        "arn:aws-cn:logs:*:*:*"
```

Comment:
Yes, I like the idea, I will incorporate that.

I also noticed some issues regarding the download of the runner RPM from within China and have to add a Chinese mirror. Because otherwise the download of the required artifacts is unfortunately simply not possible. I will add that as well if it’s ok for you.

Reply:
Yes that is fine. I assume you will add the option as a parameter with the rest of the world as default. Would be great if you can add an China example based on `runner-default` for example

Reply:
Small Update:
I made some progress today, but when you solve one problem, you run into five new ones. I have managed to incorporate a chinese mirror [repository](https://mirrors.tuna.tsinghua.edu.cn/help/gitlab-runner/) which works out quite well. However, the next thing is to install the docker-machine binary which is currently downloaded from github. It is sort of a coin flip if it goes through or not, and even when it is successful it takes more than 30 minutes to download.

Another major blocker is the docker machine runner itself. As said, I managed it to get a cluster running, however, when the docker-machine runner wants to spawn new ec2 instances, it runs into a no such host problem because it uses the URL pattern to talk to the ec2 api: ec2.<region>.amazonaws.com (ec2.cn-northwest-1.amazonaws.com) which is actually ec2.cn-northwest-1.amazonaws.com.cn in China (Ningxia Region). I will have a look if this endpoint can be configured somehow.

Reply:
Thanks for the update, did you saw this github issue? https://github.com/docker/machine/issues/4713

It seems it is possible to specify the aws endpoint.

Reply:
Finally got a setup running, however, with docker executor at first. I will make some changes to the codebase and put it all together. Furthermore, I was not able to find a reliable and "official" docker-machine binary download location, so I added a setting to override the download url :/

Reply:
Cool! Are there no available repo's in China? Could distribution via S3 work?